collector/systemd_linux.go

```diff
@@ -18,14 +18,23 @@ package collector
 import (
 	"flag"
 	"fmt"
+	"regexp"
 
 	"github.com/coreos/go-systemd/dbus"
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+var (
+	unitWhitelist = flag.String("collector.systemd.unit-whitelist", "^(.+\\.service|.+\\.socket|.+\\.mount)$", "Regexp of systemd units to whitelist.")
```

Comment:
The unit needs to both match whitelist and not match blacklist (the check is here: https://github.com/prometheus/node_exporter/pull/286/files#diff-6c687a1d682c85f158e09200041505c5R143)